draft-jmiller-json-web-proof.md

```diff
@@ -121,44 +98,27 @@ For example:
 }
 ```
 
+Every JWP algorithm must include a digest method that is used to generate a hash of the base64url serialized protected header.  The protected header cannot be selectively disclosed and the digest value MUST be included in all proof values.
+
 ## Payloads
 
-Payloads are always represented as an ordered array.  Each payload's content type is either known by context or specified in that payload's associated header.  The default content type is `application/json`, with the JSON object values being JWT claims.
+Payloads are always represented as an ordered array.  The mapping of which value is in which payload slot is out of scope of this specification.
```

Comment:
We don't have a definition of `value` nor a single type of payload, which makes this sentence very hard to understand....

In my preferred world where the payloads are JSON, this might read as:

The order and construction of specific payload objects prior to JSON serialization is out of scope for this specification as is the order in which individual payloads are assembled into the payloads.... the sentence above is trying to do too much at once...

we need to address both:

```
type Payload: JSON | ANY

type Payloads: OrderedArray<Payload>
```

both are out of scope.

Reply:
A spec layered about JWP where payloads are JSON should absolutely define this.  We'd tentatively started calling this JSON Selective Disclosure Token (JSDT), I believe.  The beauty of this PR is that it separates concerns, in the same way that concerns are separated between JWS and JWT.

Reply:
Agree that the payload contents should be addressed in a companion spec, and I also agree this wording is definitely incomplete and needs more work.

Reply:
Related to issue #8, I also made some minor improvements to this text to be more clear.